modules/rte_mis_student/rte_mis_student.module

```diff
@@ -33,6 +36,112 @@ function rte_mis_student_theme() {
+  if ($variables['view_mode'] == 'pdf' && $variables['bundle'] == 'student_details') {
+
+    $preferred_schools = $variables['content']['field_school_preferences'];
+    $preferred_school_data = [];
+    $school_info = [];
+    $rte_mis_school_config = \Drupal::config('rte_mis_school.settings');
+    foreach ($preferred_schools as $key => $value) {
+      if (is_numeric($key)) {
+        $school_details = Drupal::entityTypeManager()->getStorage('mini_node')->loadByProperties([
+          'id' => $value['#paragraph']->get('field_school_id')->getString(),
+          'type' => 'school_Details',
+        ]);
+        $school_details = reset($school_details);
+
+        if ($school_details instanceof EckEntityInterface) {
+
+          $target_id = array_column($school_details->get('field_entry_class')->getValue(), 'target_id');
+
+          // Get the entry class id.
+          $entry_class_id = $school_details->get('field_entry_class')->getValue()[0]['target_id'];
+          foreach ($target_id as $each_target_id) {
+            // Load the paragraph.
+            $entry_class_details = Paragraph::load($each_target_id);
+            if ($entry_class_details instanceof Paragraph) {
+              // User selected medium.
+              $selected_class = $variables['content']['field_class'][0]['#markup'];
+              // Load the entry class.
+              $entry_class_id = $entry_class_details->get('field_entry_class')->getString();
+              $entry_class = $rte_mis_school_config->get('field_default_options.class_level.' . $entry_class_id) ?? $entry_class_id;
+              // Match the user selected class and school entry class defined.
+              if ($entry_class == $selected_class) {
+
+                // User selected medium.
+                $selected_medium = $value['#paragraph']->get('field_medium')->getString();
+
+                // Get the label for the selected medium.
+                $medium_label = $rte_mis_school_config->get('field_default_options.field_medium.' . $selected_medium) ?? $selected_medium;
+
+                // Load the seat detail,
+                // based on the language selected by user.
+                $field_name = "field_rte_student_for_" . $selected_medium;
+                if ($entry_class_details->hasField($field_name)) {
+                  $seat = $entry_class_details->get($field_name)->getString() ?? NULL;
+                }
+
+                $term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+
+                // Get the school udise code from the term Id,
+                // provided by school details.
+                $school_udise_id = $school_details->get('field_udise_code')->getString();
+                $school_udise = $term_storage->loadByProperties([
+                  'tid' => $school_udise_id,
+                  'vid' => 'school',
+                  'status' => 1,
+                  'field_workflow' => 'school_workflow_approved',
+                ]);
```

Comment:
No need to add check here, the school's preference field has all the logic written. We can just load the term and get the name with safe check

Reply:
sure